src/Stratis.Bitcoin.Features.Api/ApiSettings.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using Stratis.Bitcoin.Configuration;
+
+namespace Stratis.Bitcoin.Features.Api
+{
+    /// <summary>
+    /// Configuration related to storage of transactions.
+    /// </summary>
+    public class ApiSettings
+    {
+        /// <summary>URI to node's API interface.</summary>
+        public Uri ApiUri { get; set; }       
+
+        private Action<ApiSettings> callback = null;
+
+        public ApiSettings()
+        {
+        }
+
+        public ApiSettings(Action<ApiSettings> callback)
+            : this()
+        {
+            this.callback = callback;
+        }
+
+        public ApiSettings(NodeSettings nodeSettings, Action<ApiSettings> callback = null)
+            : this(callback)
+        {
+            this.Load(nodeSettings);
+        }
+
+        /// <summary>
+        /// Loads the storage related settings from the application configuration.
+        /// </summary>
+        /// <param name="nodeSettings">Application configuration.</param>
+        public virtual void Load(NodeSettings nodeSettings)
```

Comment:
Not sure why this is virtual? Also can't this be private as it's only used in constructor?

Reply:
It is used in the ApiFeature to load the settings from the NodeSettings.

Reply:
Ok I see that but is that necessary as it seems like Load is also called in the constructor when this class in injected. So is the net effect Load is called twice?

Reply:
`public ApiSettings(NodeSettings nodeSettings, Action<ApiSettings> callback = null)' is not currently used so I have removed it. `Load` is called from the ApiFeature constructor.